### PR TITLE
DGarbar Implementation of LinkedList with Q (questions)

### PR DESCRIPTION
**Pull request by DGarbar**
Implementation with StreamAPI.
TODO q = question

### CODE REVIEW

File: linked-list/src/main/java/ua/procamp/LinkedList.java [hunk 1]
```diff
@@ -1,117 +1,214 @@
...
+   * @param elements elements to add
+   * @param <T> generic type
+   * @return a new list of elements the were passed as method parameters
+   */
+  public static <T> List<T> of(T... elements) {
+    LinkedList<T> tLinkedList = new LinkedList<>();
+    tLinkedList.add(elements[0]);
+    Node<T> master = tLinkedList.getNode(0);
+    for (int i = 1; i < elements.length; i++) {
+      Node<T> dep = new Node<>(elements[i]);
```
Comment (boy4uck) [inline linked-list/src/main/java/ua/procamp/LinkedList.java]: @DGarbar why don't you use `add()`?

File: linked-list/src/main/java/ua/procamp/LinkedList.java [hunk 2]
```diff
@@ -1,117 +1,214 @@
...
-     * Adds a new element to the specific position in the list. In case provided index in out of the list bounds it
-     * throws {@link IndexOutOfBoundsException}
-     *
-     * @param index   an index of new element
-     * @param element element to add
-     */
-    @Override
-    public void add(int index, T element) {
-        throw new UnsupportedOperationException("This method is not implemented yet"); // todo: implement this method
+    return Stream.iterate(head, Objects::nonNull, tNode -> tNode.next)
```
Comment (boy4uck) [inline linked-list/src/main/java/ua/procamp/LinkedList.java]: Although it looks fancy, it's overengineering.

File: linked-list/src/main/java/ua/procamp/LinkedList.java [hunk 3]
```diff
@@ -1,117 +1,214 @@
...
-        throw new UnsupportedOperationException("This method is not implemented yet"); // todo: implement this method
+  /**
+   * Retrieves an {@link Node<T>}<T></T> by its position index. In case provided index in out of the
+   * list bounds it throws {@link IndexOutOfBoundsException}
+   *
+   * @param index element index
+   * @return an  {@link Node<T>}
+   */
+  private Node<T> getNode(int index) {
+    if (index < 0 || head == null) {
```
Comment (boy4uck) [inline linked-list/src/main/java/ua/procamp/LinkedList.java]: Try using `Objects.checkIndex(index, size)` instead

File: linked-list/src/main/java/ua/procamp/LinkedList.java (hunk 1 above)
Comment (DGarbar) [reply linked-list/src/main/java/ua/procamp/LinkedList.java]: Bc I don`t have a tail, So I decided to optimize the creation.

**Pull request closed without merging**